supervisor/shared/web_workflow/web_workflow.c

```diff
@@ -329,7 +335,8 @@ void supervisor_start_web_workflow(void) {
         mdns_server_construct(&mdns, true);
         mdns.base.type = &mdns_server_type;
         if (!common_hal_mdns_server_deinited(&mdns)) {
-            common_hal_mdns_server_set_instance_name(&mdns, MICROPY_HW_BOARD_NAME);
+            char *instance_name = strndup(web_instance_name, strlen(web_instance_name));
```

Comment:
I don't think this will work right because we don't usually use `malloc()` directly. Instead, I'd just add a static variable to hold the data. Memory allocation is tricky because it can happen via the MP heap or via supervisor.

Reply:
Thanks, Scott!
Didn't think of the static variable.
Just updated the PR. Please let me know if this would work.
Thanks!